core/lib/spree/app_configuration.rb

```diff
@@ -26,6 +26,10 @@ module Spree
   class AppConfiguration < Preferences::Configuration
     # Preferences (alphabetized to more easily lookup particular preferences)
 
+    # @!attribute [rw] address_requires_phone
+    #   @return [Boolean] should phone number be required (default: +nil+)
+    preference :address_requires_phone, :boolean
```

Comment:
What about setting a default anyway? I think that being explicit with false is better.

Reply:
The reason it's nil here is so we can check to see if it's been explicitly set by the user. If it hasn't been set, a deprecation warning will display, letting the user know that this function will default to false in the future, and if they want to maintain the existing functionality, they will need to explicitly set this preference to "true". 

The idea being that we add "false" as the default when 3.0 rolls around and remove the deprecation warning.

Personally, I think the change is innocuous enough - since Solidus doesn't depend on phone numbers for anything, and it's easy enough for stores to enable the validation with this change. We're also making sure that existing stores are aware of the change, and they'll have a chance to adjust the functionality however they want. However, I understand that there are some strong opinions about this change, so I'll do whatever the core team decides here. If you'd rather this be a PR that just adds a preference that defaults to "true", then that's what we'll do.

Reply:
This makes sense Sean but we have no guarantees that devs will look at the deprecation warning before deploying so, in this form, this is considered a breaking change. I'll talk with the rest of the Core Team today in our meeting and see what they think about it.

Reply:
@seand7565 we agreed that it's better to keep the existing behavior as the default. I think what you did to force users to choose can work but if they don't we can't change their current behavior.

Reply:
Very well! I'll refactor this PR as just an added preference that defaults to true.

Reply:
Done - let me know how it looks!